src/components/TodoContentMain.tsx

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import {
+  FC, useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { todosApi } from '../api/todos-api';
+import { useTodoContext } from '../context/todoContext/useTodoContext';
+import { useErrorContext } from '../context/errorContext/useErrorContext';
+import { UpdateTodoArgs } from '../types/UpdateTodoArgs';
+
+interface TodoContentMainProps {
+  todos: Todo[],
+  tempTodo: Todo | null,
+}
+
+export const TodoContentMain: FC<TodoContentMainProps> = (props) => {
+  const { todos, tempTodo } = props;
+  const {
+    removeTodos,
+    setHandlingTodoIds,
+    handlingTodoIds,
+    updateTodos,
+  } = useTodoContext();
+  const { notifyAboutError } = useErrorContext();
+  const [editedTodoId, setEditedTodoId] = useState<number | null>(null);
+  const todoTitleRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoTitleRef.current?.focus();
+  }, [editedTodoId]);
+
+  const onRemove = async (id: number) => {
+    setHandlingTodoIds([id]);
+    try {
+      const result = await todosApi.remove([id]);
+
+      if (result) {
+        removeTodos([id]);
+      }
+    } catch {
+      notifyAboutError('Unable to delete a todo');
+    } finally {
+      setHandlingTodoIds([]);
+    }
+  };
+
+  const onEdit = async (id: number, data: UpdateTodoArgs) => {
+    try {
+      setHandlingTodoIds([id]);
+      const updatedTodos = await todosApi.update([{
+        id,
+        data,
+      }]);
+
+      updateTodos(updatedTodos);
+    } catch {
+      notifyAboutError('Unable to update a todo');
+    } finally {
+      setHandlingTodoIds([]);
+    }
+  };
+
+  const onChangeStatus = (id: number, isChecked: boolean) => {
+    onEdit(id, { completed: isChecked });
+  };
+
+  const handlerDoubleClick = (countClick: number, id: number) => {
+    if (countClick !== 2) {
+      return;
+    }
+
+    setEditedTodoId(id);
+  };
+
+  const handleTodoChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (todoTitleRef.current) {
+      todoTitleRef.current.value = event.target.value;
+    }
+  };
+
+  const handleOnBlur = () => {
+    setEditedTodoId(null);
+  };
+
+  const submit = async (
+    event: React.FormEvent<HTMLFormElement>,
+    id: number,
+  ) => {
+    event.preventDefault();
+    const trimmedTitle = todoTitleRef.current?.value.trim();
+
+    if (!trimmedTitle) {
+      onRemove(id);
+
+      return;
+    }
+
+    if (todoTitleRef.current) {
+      await onEdit(id, { title: todoTitleRef.current.value });
```

Comment:
You are using `todoTitleRef.current.value` directly to update the Todo. It's better to use `trimmedTitle` that you created at line 94, in order to ensure that the updated title is the trimmed version.